core/dbt/parser/partial.py

```diff
@@ -211,6 +223,9 @@ def update_in_saved(self, file_id):
     # Updated schema files should have been processed already.
     def update_mssat_in_saved(self, new_source_file, old_source_file):
 
+        if self.already_scheduled_for_parsing(old_source_file):
```

Comment:
Could we dry these checks up a little by putting a single check in `update_in_saved`?  AFAICT that's the only place these functions are called.

Reply:
That would work, but I'm hoping to clean up the factoring of this code in the future and reuse these functions, so I'm going to leave them as separate calls for now.